fsspec/tests/test_spec.py

```diff
@@ -744,6 +745,31 @@ def test_cache():
     assert len(DummyTestFS._cache) == 0
 
 
+def test_cache_not_pickled(server):
```

Comment:
Alright, it took some back and forth debugging via CI, but I've got this test working with `HTTPFile` now. @martindurant let me know what you think